### PR TITLE
Add pmc publisher prototype

### PR DESCRIPTION
**Pull request by gabrielhartmann**
The PMC publisher logs LLC misses and LLC hits for every process that is
part of a Titus task

### CODE REVIEW

File: titus_isolate/metrics/bpf.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from __future__ import print_function
+from bcc import BPF, PerfType, PerfHWConfig
```
Comment (rostyboost) [inline titus_isolate/metrics/bpf.py]: don't forget to add it to `requirements.txt`

File: titus_isolate/metrics/bpf.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    struct key_t key = {};
+    get_key(&key);
+
+    ref_count.increment(key, ctx->sample_period);
+
+    return 0;
+}
+"""
+
+sample_period = 100
```
Comment (rostyboost) [inline titus_isolate/metrics/bpf.py]: nit: upper case, and document unit

File: titus_isolate/metrics/bpf.py (hunk 1 above)
Comment (gabrielhartmann) [reply titus_isolate/metrics/bpf.py]: Sadly can't be added to `requirements.txt`. Must be installed with `apt-get install python3-bcc` on the target host.

File: titus_isolate/metrics/bpf.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+b = BPF(text=bpf_text)
+try:
+    b.attach_perf_event(
+        ev_type=PerfType.HARDWARE, ev_config=PerfHWConfig.CACHE_MISSES,
+        fn_name="on_cache_miss", sample_period=sample_period)
+    b.attach_perf_event(
+        ev_type=PerfType.HARDWARE, ev_config=PerfHWConfig.CACHE_REFERENCES,
+        fn_name="on_cache_ref", sample_period=sample_period)
+except:
+    print("Failed to attach to a hardware event. Is this a virtual machine?")
```
[the remark this answers is not in the document]
Comment (gabrielhartmann) [reply titus_isolate/metrics/bpf.py]: thanks.  Print is no good.

File: titus_isolate/metrics/bpf.py (hunk 2 above)
Comment (gabrielhartmann) [reply titus_isolate/metrics/bpf.py]: done

**Pull request merged**